### DISCUSSION TITLE
✅ Pascal Case Names used in `makeCleanId`

### DISCUSSION THREAD

**Discussion by shawncarr** [RobinTail/express-zod-api#1377] [Ideas]
Currently, if you have a `GET`  path such as `/companys/:companyId/users/:userId` and you generate a swagger or client resulting name will be `GetCompaniesCompanyidUsersUserid`.

IMHO it might be better to pascal case each entry so that easier to read so that it would be `GetCompaniesCompanyIdUsersUserId`

```
// NOTE this was just the first toPascalCase result I found
const toPascalCase = (input: string): string => {
  return (input.match(/[A-Z]{2,}(?=[A-Z][a-z]+[0-9]*|\b)|[A-Z]?[a-z]+[0-9]*|[A-Z]|[0-9]+/g) ?? [])
    .map((x) => x.charAt(0).toUpperCase() + x.slice(1).toLowerCase())
    .join('');
};

export const makeCleanId = (path: string, method: string, suffix?: string) => {
  return [method]
    .concat(path.split('/'))
    .concat(suffix || [])
    .flatMap((entry) => entry.split(/[^A-Z0-9]/gi))
    .map((entry) => toPascalCase(entry))
    .join('');
};
```

**Comment by RobinTail** [comment]
I agree that it would be better.
Would it be possible to achieve it without running such complex regular expressions, @shawncarr ?

**Reply by shawncarr** [reply]
I'd have to research it. My only goal was not to add another library just for that so I found the first article that would do the job for the example implementation.

**Reply by RobinTail** [reply]
Ok.
Perhaps besides splitting by slashes, we could also split by uppercase letters, getting words out of it. Then ensure each word has first letter capitalized and concatenate back.

**Comment by RobinTail** [comment]
@shawncarr , please check this out and let me know what you think:
https://github.com/RobinTail/express-zod-api/pull/1381

**Reply by shawncarr** [reply]
Nice and clean so I like it.

**Reply by RobinTail** [reply]
Thank you, @shawncarr .
The improvement released in version 15.2.0 🚀
